Documentation: Roles and Permissions and Groups (#9005)
opf/openproject: docs/system-admin-guide/users-permissions/groups/README.md

# Manage Groups

<div class="glossary">

> A **Group** is defined as a list of users which can be assigned to a project with a selected role. New groups can be defined in Administration -> Users & Permissions -> Groups.

Reviewer: ```suggestion
A **Group** is defined as a list of users which can be assigned to a project with a selected role. New groups can be defined in *-> Administration -> Users & Permissions -> Groups*.
```

Author: Accepted

<div class="glossary">
**Group** is defined as a list of users which can be assigned to a project with a certain role. New groups can be defined in the admin settings (Modules > Administration).
  </div>
A **Group** is defined as a list of users which can be assigned to a project with a selected role. New groups can be defined in Administration -> Users & Permissions -> Groups.
</div>

> ## Add a new group

Reviewer: I'm not sure but the "group" should be deleted too, I guess?

Author: I created a new section for "delete groups", and moved some sentences to that section, as noted below.

> OpenProject allows creating tailored project member **groups**, which  grant additional permissions to individual users within a project. Instead of adding individual users to a project you can add a user group, e.g. Marketing. You can edit the existing groups and create new ones, adding and removing users.

Reviewer: ```suggestion
OpenProject allows creating tailored project member **groups**, which  grant additional permissions to individual users within a project. Instead of adding individual users to a project you can add a user group, e.g. Marketing. You can edit existing groups and create new ones or add and remove users.
```

Author: Accepted. Added "delete groups" in list

![create groups](image-20200211142053557.png)

![no existing groups](image-20200115164118182.png)

> Click the green **+ Group** button to create a new group.

Reviewer: please put this sentence above of the two screenshots (in the docs the instruction is always above the screenshot in which the respective button can be seen [I will add this to the guide for writing documentation]

Author: Accepted - looking forward to the new guide

![new group](image-20200115164328352.png)

> Give a name for your new group, which must be unique, and click the blue **Create** button. The [project members list](../../getting-started/invite-members) shows groups and users in the selection field, which could affect your choice of names.

Reviewer: Good idea to add the fact that group names must be unique!

When trying this I found that I can assign names to a group that have already been used as user name (for a user). I think referencing the project member list of individual projects could confuse the readers. I would leave the second sentence out: 
```suggestion
Give a name for your new group, which must be unique, and click the blue **Create** button. 
```

Author: Accepted

## Add users to a group, edit or remove groups

> You can add users to a group and edit a group by clicking on the **name of the group** (1) and  remove the group by clicking on **delete** icon in the respective line (2). Deleting a group will remove the role of the group in a project; users with no other roles are removed from the project.

Reviewer: suggestion:
```suggestion
You can add users to a group and edit a group by clicking on the **name of the group** (1) and remove the group by clicking on **delete** icon in the respective line (2). Deleting a group will remove the role of the group in a project; users with no other roles (i.e. who were only added as member of this group and not individually) are removed from the project.
```

Author: I have moved the last sentence to the new "Delete Groups" section

![Sys-admin-edit-groups](Sys-admin-edit-groups.png)

> After selecting the editing mode, you can **change the group name**, **add or remove group members** and **assign groups to projects**.

Reviewer: I like it.
One small change:
```suggestion
After clicking one a group's name, you can **change the group name**, **add or remove group members** and **assign groups to projects**.
```

Author: Accepted

![edit groups](image-20200115164847858.png)

### Add users to a group

Click the **Users** tab. Select the users you want to add to this group from the "New user" drop-down list. Click the blue **Add** button. Users already in the group are not shown in the list.

> Adding a user to a group also adds them to projects using that group.

Reviewer: ```suggestion
Adding a user to a group also adds them to projects in which this group was assigned a [role](../roles-permissions) (e.g. member).
```
What do you think?

Author: Accepted

![Sys-admin-add-users-groups](Sys-admin-add-users-groups.gif)

### Add projects to a group

Click the **Projects** tab. Select the projects you want to add this group to from the "New project" drop-down list. Tick the roles that you want the group to have. Click the blue **Add** button. Users in the group are added to that project's members using the role selected.

> ## Impact of groups on project members

Reviewer: Good idea! I like it!
I would suggest to move this section partly to the user-oriented part of the documentation (in this case the Getting started guide) as this is more relevant for project admins than for system admins. I'd suggest leaving lines 58 to 62 (the first 5 bullet points) out here. I already added them to the respective section about project members.

Author: Accepted. I also removed the last two, as these are covered in previous sections.

> Groups have the following impact on a [project members list](../../getting-started/invite-members) and the [user details](../users). Changes in groups, project members or users may affect the other two.

Reviewer: ```suggestion
Groups have the following impact on a [project members list](../../../getting-started/invite-members) and the [user details](../users). Changes in groups, project members or users may affect the other two.
```

Author: Accepted

* the group shows as a separate line on the project members list
* the group members cannot be removed from the members list (no delete icon)
* adding a group with members already in a project member list has the group role added to their project roles
* a project member belonging to a group can have additional roles added individually
* the group role on an individual project member cannot be removed

> * adding a user to a group adds that user to the member list of all projects using that group

Reviewer: ```suggestion
* adding a user to a group adds that user to the member list of *all* projects using that group
```

Author: Accepted

> * deleting the group removes the role from the members of any project using that group. If the user has no other roles it is removed from the project members list

Reviewer: ```suggestion
* deleting a group removes the role from the members of any project using that group. If the user has no other roles in a project it is removed from the respective project completely

Find out more about the behavior of groups as project members from a project admin's perspective [here](../../../getting-started/invite-members/#behavior-of-groups-as-project-members
```

Author: Accepted
